File: gherkinexecutor/Feature_Examples/ValueValid.py
import json

class ValueValid:
    def __init__(self, value="0", valid="false", notes=""):
        self.value = value
        self.valid = valid
        self.notes = notes

    def __eq__(self, other):
        if self is other:
            return True
        if other is None or not isinstance(other, ValueValid):
            return False
        if self.value != "?DNC?" and other.value != "?DNC?" and self.value != other.value:
            return False
        if self.valid != "?DNC?" and other.valid != "?DNC?" and self.valid != other.valid:
            return False
        if self.notes != "?DNC?" and other.notes != "?DNC?" and self.notes != other.notes:
            return False
        return True

    def __str__(self):
        return f"ValueValid {{value = {self.value} valid = {self.valid} notes = {self.notes} }}\n"

    def to_json(self):
        return json.dumps({"value": self.value, "valid": self.valid, "notes": self.notes})

    @staticmethod
    def from_json(json_str):
        data = json.loads(json_str)
        instance = ValueValid()
        instance.value = data.get("value", "0")
        instance.valid = data.get("valid", "false")
        instance.notes = data.get("notes", "")
        return instance

    @staticmethod
    def list_to_json(list_of_value_valids):
        return json.dumps([value_valid.to_json() for value_valid in list_of_value_valids])

    @staticmethod
    def list_from_json(json_str):
        list_of_value_valids = json.loads(json_str)
        return [ValueValid.from_json(json.dumps(item)) for item in list_of_value_valids]

    class Builder:
        def __init__(self):
            self.value = "0"
            self.valid = "false"
            self.notes = ""

        def setvalue(self, value):
            self.value = value
            return self

        def setvalid(self, valid):
            self.valid = valid
            return self

        def setnotes(self, notes):
            self.notes = notes
            return self

        def set_compare(self):
            self.value = "?DNC?"
            self.valid = "?DNC?"
            self.notes = "?DNC?"
            return self

        def build(self):
            return ValueValid(self.value, self.valid, self.notes)

    def to_value_valid_internal(self):
        return ValueValidInternal(self.value, self.valid.lower() == "true", self.notes)

# Assuming ValueValidInternal class is defined elsewhere

File: gherkinexecutor/Feature_Examples/ValueValid.java
package gherkinexecutor.Feature_Examples;
import java.util.*;
class ValueValid{
    String value = "0";
    String valid = "false";
    String notes = "";
    public ValueValid() { }
    public ValueValid(
        String value
        ,String valid
        ,String notes
        ){
        this.value = value;
        this.valid = valid;
        this.notes = notes;
        }
    @Override
    public boolean equals (Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass())
             return false;
        ValueValid _ValueValid = (ValueValid) o;
            boolean result = true;
         if (
             !this.value.equals("?DNC?")
                && !_ValueValid.value.equals("?DNC?"))
                if (! _ValueValid.value.equals(this.value)) result = false;
         if (
             !this.valid.equals("?DNC?")
                && !_ValueValid.valid.equals("?DNC?"))
                if (! _ValueValid.valid.equals(this.valid)) result = false;
         if (
             !this.notes.equals("?DNC?")
                && !_ValueValid.notes.equals("?DNC?"))
                if (! _ValueValid.notes.equals(this.notes)) result = false;
             return result;  }
    public static class Builder {
        private String value = "0";
        private String valid = "false";
        private String notes = "";
        public Builder value(String value) {
            this.value = value;
            return this;
            }
        public Builder valid(String valid) {
            this.valid = valid;
            return this;
            }
        public Builder notes(String notes) {
            this.notes = notes;
            return this;
            }
        public Builder  setCompare() {
            value = "?DNC?";
            valid = "?DNC?";
            notes = "?DNC?";
            return this;
            }
        public ValueValid build(){
             return new ValueValid(
                 value
                 ,valid
                 ,notes
                );   } 
        } 
    @Override
    public String toString() {
        return "ValueValid {"
        +"value = " + value + " "
        +"valid = " + valid + " "
        +"notes = " + notes + " "
            + "} " + "\n"; }  
    public String toJson() {
        return " {"
        +""+"value:" + "\"" + value + "\""
        +","+"valid:" + "\"" + valid + "\""
        +","+"notes:" + "\"" + notes + "\""
            + "} " + "\n"; }  
        public static ValueValid fromJson(String json) {
              ValueValid instance = new ValueValid();

              	json = json.replaceAll("\\s", "");
                String[] keyValuePairs = json.replace("{", "").replace("}", "").split(",");

                // Iterate over the key-value pairs
                for (String pair : keyValuePairs) {
                    // Split each pair by the colon
                    String[] entry = pair.split(":");

                    // Remove the quotes from the key and value
                    String key = entry[0].replace("\"", "").trim();
                    String value = entry[1].replace("\"", "").trim();


          // Assign the value to the corresponding field
                    switch (key) {
              case "value":
                  instance.value = value;
                  break;
              case "valid":
                  instance.valid = value;
                  break;
              case "notes":
                  instance.notes = value;
                  break;
        				default:
        				    System.err.println("Invalid JSON element " + key);
                    }
                }
                return instance;
            }


             public static String listToJson(List<ValueValid> list) {
                 StringBuilder jsonBuilder = new StringBuilder();
                 jsonBuilder.append("[");

                 for (int i = 0; i < list.size(); i++) {
                     jsonBuilder.append(list.get(i).toJson());
                     if (i < list.size() - 1) {
                         jsonBuilder.append(",");
                     }
                 }

                 jsonBuilder.append("]");
                 return jsonBuilder.toString();
             }

             public static List<ValueValid> listFromJson(String json) {
                    List<ValueValid> list = new ArrayList<>();
            		json = json.replaceAll("\\s", "");
                    String[] jsonObjects = json.replace("[", "").replace("]", "").split("\\},\\{");

                    for (String jsonObject : jsonObjects) {
                        jsonObject = "{" + jsonObject.replace("{", "").replace("}", "") + "}";
                        list.add(ValueValid.fromJson(jsonObject));
                    }
                    return list;
                }

    ValueValidInternal toValueValidInternal() {
        return new ValueValidInternal(
         value
        , Boolean.parseBoolean(valid)
        , notes
        ); }
    }
